app/src/main/java/de/danoeh/antennapod/fragment/DiscoveryFragment.java

```diff
@@ -95,16 +105,17 @@ public void onCreate(Bundle savedInstanceState) {
     }
 
     @Override
-    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
-                             Bundle savedInstanceState) {
+    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
         // Inflate the layout for this fragment
         View root = inflater.inflate(R.layout.fragment_itunes_search, container, false);
         gridView = root.findViewById(R.id.gridView);
         adapter = new ItunesAdapter(getActivity(), new ArrayList<>());
         gridView.setAdapter(adapter);
 
-        MaterialToolbar toolbar = root.findViewById(R.id.toolbar);
+        toolbar = root.findViewById(R.id.toolbar);
         toolbar.setNavigationOnClickListener(v -> getParentFragmentManager().popBackStack());
+        toolbar.inflateMenu(R.menu.countries_menu);
+        toolbar.setOnMenuItemClickListener(this);
```

Comment:
If you close and re-open the app, the checked state is not kept updated

Reply:
fixed